llvm/test/Transforms/InstCombine/sink_instruction.ll

```diff
@@ -272,3 +272,118 @@ abort:
   call void @abort()
   unreachable
 }
+
+; Loads marked invariant can be sunk across critical edges.
```

Comment:
What do critical edges have to do with this? It's about sinking past potential memory writes.